js/modules/loki_rss_api.js

```diff
@@ -0,0 +1,163 @@
+/* eslint-disable no-await-in-loop */
+/* eslint-disable no-loop-func */
+/* global log, window, textsecure, ConversationController */
+
+const EventEmitter = require('events');
+const nodeFetch = require('node-fetch');
+
+const RSS_FEED = 'https://loki.network/feed/';
+const CONVO_ID = 'rss://loki.network/feed/';
+const PER_MIN = 60 * 1000;
+const PER_HR = 60 * PER_MIN;
+const RSS_POLL_EVERY = 1 * PER_HR; // once an hour
+
+function xml2json(xml) {
+  try {
+    let obj = {};
+    if (xml.children.length > 0) {
+      for (let i = 0; i < xml.children.length; i += 1) {
+        const item = xml.children.item(i);
+        const { nodeName } = item.nodeName;
+
+        if (typeof obj[nodeName] === 'undefined') {
+          obj[nodeName] = xml2json(item);
+        } else {
+          if (typeof obj[nodeName].push === 'undefined') {
+            const old = obj[nodeName];
+
+            obj[nodeName] = [];
+            obj[nodeName].push(old);
+          }
+          obj[nodeName].push(xml2json(item));
+        }
+      }
+    } else {
+      obj = xml.textContent;
+    }
+    return obj;
+  } catch (e) {
+    log.error(e.message);
+  }
+  return {};
+}
+
+// hate duplicating this here...
+const friendRequestStatusEnum = Object.freeze({
+  // New conversation, no messages sent or received
+  none: 0,
+  // This state is used to lock the input early while sending
+  pendingSend: 1,
+  // Friend request sent, awaiting response
+  requestSent: 2,
+  // Friend request received, awaiting user input
+  requestReceived: 3,
+  // We did it!
+  friends: 4,
+  // Friend Request sent but timed out
+  requestExpired: 5,
+});
+
+class LokiRssAPI extends EventEmitter {
+  constructor() {
+    super();
+    // properties
+    this.groupId = CONVO_ID;
+    this.feedTimer = null;
+    this.conversationSetup = false;
+    // initial set up
+    this.getFeed();
+  }
+
+  setupConversation() {
+    // only run once
+    if (this.conversationSetup) return;
+    // wait until conversations are loaded
+    if (ConversationController._initialFetchComplete) {
+      const conversation = ConversationController.getOrCreate(
+        this.groupId,
+        'group'
+      );
+      conversation.setFriendRequestStatus(friendRequestStatusEnum.friends);
```

Comment:
Since you're just using the friend status, it might make sense to just define a single const above instead of copying the whole `friendRequestStatusEnum`

Reply:
yea I'd really like a way to define this so we can load it on the browser and node.js side, so we only have one place to maintain it